themes/default/content/blog/typescript-versions/index.md

```diff
@@ -0,0 +1,97 @@
+# the time portion of the date value; posts are sorted in descending order by
+# date/time.
+date: 2024-04-15T10:28:28+02:00
+
+# The draft setting determines whether a post is published. Set it to true if
+# you want to be able to merge the post without publishing it.
+draft: false
+
+# Use the meta_desc property to provide a brief summary (one or two sentences)
+# of the content of the post, which is useful for targeting search results or
+# social-media previews. This field is required or the build will fail the
+# linter test. Max length is 160 characters.
+meta_desc: The latest version of Pulumi's Node.js SDK now lets you easily pick the version of TypeScript to use.
+
+# The meta_image appears in social-media previews and on the blog home page. A
+# placeholder image representing the recommended format, dimensions and aspect
+# ratio has been provided for you.
+meta_image: meta.png
+
+# At least one author is required. The values in this list correspond with the
+# `id` properties of the team member files at /data/team/team. Create a file for
+# yourself if you don't already have one.
+authors:
+    - julien-poissonnier
+
+# At least one tag is required. Lowercase, hyphen-delimited is recommended.
+tags:
+    - typescript
+    - nodejs
+    - user-experience
+
+# See the blogging docs at https://github.com/pulumi/pulumi-hugo/blob/master/BLOGGING.md
+# for details, and please remove these comments before submitting for review.
+---
+
+Pulumi allows you to use the best features of programming languages to build your infrastructure while ensuring compatibility with your current development environments. To this end Pulumi provides a seamless experience when authoring TypeScript programs, managing the compilation of your program for you. With the latest release of our Node.js SDK, we've made this even more flexible by making it easier than ever to chose the version of TypeScript you want to use.
+
+<!--more-->
+
+Each new release of TypeScript comes with new features and improved type checking, as well as support for the latest JavaScript features. Using the latest version of TypeScript provides you with the maximum flexibility for creating types and gives you more confidence that your program will behave as expected.
+
+Until now, Pulumi's seamless experience was limited to using Pulumi's builtin TypeScript version 3.8. If you wanted to use a more recent version, you had to introduce a build step to your project and let Pulumi run in [plain Javascript mode](https://www.pulumi.com/docs/languages-sdks/javascript/#disabling-built-in-typescript-support). With the release of the [Pulumi Node.js SDK 3.113.0](https://www.npmjs.com/package/@pulumi/pulumi) you can now use any version of TypeScript by adding it as a dependency to your project's `package.json` file.
+
+To upgrade your project to the latest versions of TypeScript and Pulumi, run:
+
+```bash
+npm add typescript@^5.4.5 @pulumi/pulumi@^3.113.0
+```
+
```

Comment:
Perhaps "Adding TypeScript as a dependency to your project is optional, as TypeScript and ts-node are optional peer dependencies of the Node.js SDK. This makes it easy to use a single version of TypeScript or ts-node within monorepos. To ensure backward compatibility, Pulumi ships with bundled versions of TypeScript 3.8 and ts-node 7, and falls back to these versions if they are not installed in your project's node modules."